### PR TITLE
feat: Outbound dispatcher

### LINKED ISSUE #259: Outbound dispatcher
Created based on comments in https://github.com/hyperledger/aries-framework-go/pull/240#discussion_r324789454

### PR DESCRIPTION
**Pull request by fqutishat**
- Add Outbound dispatcher implementation

closes #259

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/didcomm/dispatcher/outbound.go [hunk 1]
```diff
@@ -7,23 +7,43 @@ SPDX-License-Identifier: Apache-2.0
...
+		if v.Accept(des.ServiceEndpoint) {
+			bytes, err := json.Marshal(msg)
+			if err != nil {
+				return fmt.Errorf("failed marshal to bytes: %w", err)
+			}
+			packedMsg, err := o.wallet.PackMessage(&wallet.Envelope{Message: bytes, FromVerKey: senderVerKey, ToVerKeys: des.RecipientKeys})
+			if err != nil {
+				return fmt.Errorf("failed to pack msg: %w", err)
+			}
+			// TODO should we return respData from send
```
Comment (llorllale) [inline pkg/didcomm/dispatcher/outbound.go]: Note: DIDComm is simplex  by default

File: pkg/didcomm/protocol/didexchange/service.go [hunk 2]
```diff
@@ -425,7 +425,7 @@ func (s *Service) SendExchangeResponse(exchangeResponse *Response, destination s
 func (s *Service) marshalAndSend(data interface{}, errorMsg, destination string) (string, error) {
 	// TODO an outboundtransport implementation should be selected based on the destination's URL.
 	// TODO add Destination struct
-	return "", s.outboundDispatcher.Send(data, nil)
+	return "", s.outboundDispatcher.Send(data, "", nil)
```
Comment (talwinder50) [inline pkg/didcomm/protocol/didexchange/service.go]: We dont need this function. If your change is going before i think you should just remove it.

Comment (fqutishat) [reply pkg/didcomm/protocol/didexchange/service.go]: ok I will remove it

Comment (fqutishat) [reply pkg/didcomm/protocol/didexchange/service.go]: If I remove it will effect other functions

Comment (fqutishat) [reply pkg/didcomm/protocol/didexchange/service.go]: maybe you remove it in your pr

Comment (talwinder50) [reply pkg/didcomm/protocol/didexchange/service.go]: okay i have already removed it in my pr  and made required changes. Ignore it for now

**Pull request merged**